### PR TITLE
edit elementwise_mul doublegrad inplace

### PR DESCRIPTION
**Pull request by danleifeng**
fix elementwise mul double_grad kernel
inpalce will occur wrong in unittest when size(ddx) != size(ddout)

### CODE REVIEW

File: paddle/fluid/operators/elementwise/elementwise_mul_op.h [hunk 1]
```diff
@@ -172,34 +172,49 @@ class ElementwiseMulDoubleGradKernel : public framework::OpKernel<T> {
...
-          MulGradDX<T>(), MulGradDY<T>());
-      default_elementwise_mul<DeviceContext, T>(ctx, &ddx_safe, y, ddout);
-
-      auto& place =
-          *ctx.template device_context<DeviceContext>().eigen_device();
-      auto ddout_t = framework::EigenVector<T>::Flatten(*ddout);
-      auto ddout_tmp_t = framework::EigenVector<T>::Flatten(*ddout_tmp);
-      ddout_t.device(place) = ddout_t + ddout_tmp_t;
-      default_elementwise_mul<DeviceContext, T>(ctx, dout, &ddy_safe, dx);
+      // when size(x) < size(y), ddout can't use memory of dx using inplace.
```
Comment (zhiqiu) [inline paddle/fluid/operators/elementwise/elementwise_mul_op.h]: Would it be better to say "size(ddout) > size(ddx), ddout can't use memory of ddx using inplace"？

Comment (danleifeng) [reply paddle/fluid/operators/elementwise/elementwise_mul_op.h]: yes! thanks!

File: paddle/fluid/operators/elementwise/elementwise_mul_op.h [hunk 2]
```diff
@@ -172,34 +172,49 @@ class ElementwiseMulDoubleGradKernel : public framework::OpKernel<T> {
...
+        ElemwiseGradCompute<DeviceContext, T, MulGradDX<T>, MulGradDY<T>>(
+            ctx, ddx_safe, ddy_safe, *dout, *dout, axis, dx, dy, MulGradDX<T>(),
+            MulGradDY<T>());
+        Tensor ddout_tmp;
+        ddout_tmp.mutable_data<T>(ddout->dims(), ctx.GetPlace());
+        default_elementwise_mul<DeviceContext, T>(ctx, y, &ddx_safe, ddout);
+        default_elementwise_mul<DeviceContext, T>(ctx, &ddy_safe, x,
+                                                  &ddout_tmp);
+        auto& place =
+            *ctx.template device_context<DeviceContext>().eigen_device();
```
Comment (ceci3) [inline paddle/fluid/operators/elementwise/elementwise_mul_op.h]: put place outside of if(ddout->numel() > ddx->numel()) to reduce code repeat?

Comment (danleifeng) [reply paddle/fluid/operators/elementwise/elementwise_mul_op.h]: done.thanks

**Review by ceci3 — APPROVED** [review]
LGTM

**Pull request merged**